app/models/rbac.rb

```diff
@@ -463,16 +461,32 @@ def self.search(options = {})
     ids_clause             = nil
     target_ids             = nil
 
-    unless targets.empty?
+    if targets.nil? || targets.kind_of?(Array) && targets.empty?
+      scope = apply_scope(klass, scope)
+    elsif targets.kind_of?(Array) && !targets.empty?
       if targets.first.kind_of?(Numeric)
-        target_ids       = targets
+        target_ids = targets
       else
         target_ids       = targets.collect(&:id)
         klass            = targets.first.class.base_class unless klass.respond_to?(:find)
         results_format ||= :objects
       end
+      scope = apply_scope(klass, scope)
 
       ids_clause = ["#{klass.table_name}.id IN (?)", target_ids] if klass.respond_to?(:table_name)
+    else # targets is a scope / class
+      targets = targets.to_s.constantize if targets.kind_of?(String) || targets.kind_of?(Symbol)
+      targets = targets.all if targets < ActiveRecord::Base
```

Comment:
there are a number of places that pass in a string for the class name.
From a hash stored in the database

longer term it would be better not to have this